site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,177 @@
+
+{% include_codefile 'demo/initial/biz-e-corp/src/main.ts' %}
+
+Now, we need to update `main` to construct and populate the `ApplicationContext` as well as register that context so that it is accessible to the rest of the application.
+
+{% instruction 'Import the `ApplicationContext` module and add this code to `main`:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' lines:10-29 %}
+
+{% aside 'Loading data' %}
+In a real-world application, this data would probably be loaded via a call to a web-service or a local datastore. To learn more, take a look at the [data stores tutorial](../comingsoon.html).
+{% endaside %}
+
+This is the same data that we used in the `App` module to initialize the `WorkerProperties`, but it is now in a more appropriate location. In general, the `main` module of an application should be concerned with initializing application-wide state. Also, as we mentioned earlier, the `App` class only needed the `WorkerProperties` because its children needed them.
+
+Now that we have the `ApplicationContext`, it is time to allow it to be used by widgets. To do that, we need to create an `Injector` and add it to the application's central widget registry. `Injectors` are created using the `Injector` factory function from `@dojo/widget-core`.
+
+{% instruction 'Add the following imports to the `main` module.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' lines:2-3 %}
+
+{% instruction 'Now, add the following after the `applicationContext` declaration.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' line:31 %}
+
+This statement does two things. First, it uses the `Injector` factory function to wrap the `applicationContext` with the `BaseInjector`, a lightweight component that allows any object that extends `Evented` to be injected via a `Container`, which we will talk about in the next section. This statement also registers the `Injector` with the application's central registry. The registry provides a way to register a widget via a label, making it accessible to other parts of the application. You can learn more in the [registry tutorial](../comingsoon.html).
+
+Now that the `Injector` is defined and registered, it is time to create the components that will use it. In the next section, we will create a special type of non-visual widget called a `Container` that will allow state to be injected into the `WorkerForm` and `WorkerContainer` widgets.
+
+{% section %}
+
+## Creating state containers
+
+{% task 'Create `Containers` that will allow state to be injected into widgets' %}
+
+On their own, `Injectors` are not able to help us very much because widgets expect state to be pushed into them (via their properties), but state must be pulled from an `Injector`. `Containers` are designed to coordinate the injection - they pull state from injectors and push it into widgets.
```

Comment:
"On their own, `Injectors` are not able to help us very much because widgets expect state to be pushed into them (via their properties), but state must be pulled from an `Injector`. `Containers` are designed to coordinate the injection - they pull state from injectors and push it into widgets." -> "On their own, `Injectors` are limited in their capabilities because widgets expect state to be pushed into them (via their properties), but state must be pulled from an `Injector`. `Containers` are designed to coordinate the injection - they pull state from injectors and push it into widgets."